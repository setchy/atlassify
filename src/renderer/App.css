html,
body,
#root {
  height: 100%;
  -webkit-user-select: none;
}

html::-webkit-scrollbar,
div.flex-grow::-webkit-scrollbar {
  width: 10px;
}

html::-webkit-scrollbar-thumb,
div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #c1c1c1;
  border-radius: 10px;
}

html::-webkit-scrollbar-thumb:hover,
div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #a8a8a8;
  border-radius: 10px;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  cursor: default;
}

/**
 * Set emoji size according to surrounding text.
 * Ref: https://github.com/jdecked/twemoji?tab=readme-ov-file#inline-styles
 */
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
