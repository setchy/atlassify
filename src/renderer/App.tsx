import { useContext } from 'react';
import {
  Navigate,
  Route,
  HashRouter as Router,
  Routes,
  useLocation,
} from 'react-router-dom';

import { AppContext, AppProvider } from './context/App';
import { AccountsRoute } from './routes/Accounts';
import { FiltersRoute } from './routes/Filters';
import { LandingRoute } from './routes/Landing';
import { LoginRoute } from './routes/Login';
import { NotificationsRoute } from './routes/Notifications';
import { SettingsRoute } from './routes/Settings';

import './App.css';
import { AppLayout } from './components/layout/AppLayout';

function RequireAuth({ children }) {
  const { isLoggedIn } = useContext(AppContext);
  const location = useLocation();

  return isLoggedIn ? (
    children
  ) : (
    <Navigate to="/landing" replace state={{ from: location }} />
  );
}

export const App = () => {
  return (
    <AppProvider>
      <Router>
        <AppLayout>
          <Routes>
            <Route
              path="/"
              element={
                <RequireAuth>
                  <NotificationsRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/filters"
              element={
                <RequireAuth>
                  <FiltersRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/settings"
              element={
                <RequireAuth>
                  <SettingsRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/accounts"
              element={
                <RequireAuth>
                  <AccountsRoute />
                </RequireAuth>
              }
            />
            <Route path="/landing" element={<LandingRoute />} />
            <Route path="/login" element={<LoginRoute />} />
          </Routes>
        </AppLayout>
      </Router>
    </AppProvider>
  );
};
