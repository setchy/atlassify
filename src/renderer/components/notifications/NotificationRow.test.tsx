import { fireEvent, render, screen } from '@testing-library/react';
import { mockSingleAtlassifyNotification } from '../../__mocks__/notifications-mocks';
import {
  mockAtlassianCloudAccount,
  mockAuth,
  mockSettings,
} from '../../__mocks__/state-mocks';
import { AppContext } from '../../context/App';
import type { ReadState } from '../../types';
import * as comms from '../../utils/comms';
import * as links from '../../utils/links';
import { NotificationRow } from './NotificationRow';

describe('renderer/components/notifications/NotificationRow.tsx', () => {
  jest.spyOn(links, 'openNotification');
  jest.spyOn(comms, 'openExternalLink').mockImplementation();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render itself & its children - group by date', async () => {
    jest
      .spyOn(global.Date, 'now')
      .mockImplementation(() => new Date('2024').valueOf());

    const props = {
      notification: mockSingleAtlassifyNotification,
      account: mockAtlassianCloudAccount,
    };

    const tree = render(
      <AppContext.Provider
        value={{
          settings: { ...mockSettings, groupNotificationsByProduct: false },
        }}
      >
        <NotificationRow {...props} />
      </AppContext.Provider>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render itself & its children - group by products', async () => {
    jest
      .spyOn(global.Date, 'now')
      .mockImplementation(() => new Date('2024').valueOf());

    const props = {
      notification: mockSingleAtlassifyNotification,
      account: mockAtlassianCloudAccount,
    };

    const tree = render(
      <AppContext.Provider
        value={{
          settings: { ...mockSettings, groupNotificationsByProduct: true },
        }}
      >
        <NotificationRow {...props} />
      </AppContext.Provider>,
    );

    expect(tree).toMatchSnapshot();
  });

  describe('notification interactions', () => {
    it('should open a notification in the browser - click', () => {
      const markNotificationsReadMock = jest.fn();

      const props = {
        notification: mockSingleAtlassifyNotification,
        account: mockAtlassianCloudAccount,
      };

      render(
        <AppContext.Provider
          value={{
            settings: mockSettings,
            markNotificationsRead: markNotificationsReadMock,
            auth: mockAuth,
          }}
        >
          <NotificationRow {...props} />
        </AppContext.Provider>,
      );

      fireEvent.click(screen.getByTestId('notification-row'));

      expect(links.openNotification).toHaveBeenCalledTimes(1);
      expect(markNotificationsReadMock).toHaveBeenCalledTimes(1);
    });

    it('should open a notification in the browser - key down', () => {
      const markNotificationsReadMock = jest.fn();

      const props = {
        notification: mockSingleAtlassifyNotification,
        account: mockAtlassianCloudAccount,
      };

      render(
        <AppContext.Provider
          value={{
            settings: mockSettings,
            markNotificationsRead: markNotificationsReadMock,
            auth: mockAuth,
          }}
        >
          <NotificationRow {...props} />
        </AppContext.Provider>,
      );

      fireEvent.keyDown(screen.getByTestId('notification-row'));

      expect(links.openNotification).toHaveBeenCalledTimes(1);
      expect(markNotificationsReadMock).toHaveBeenCalledTimes(1);
    });

    it('should open a notification in the browser - delay notification setting enabled', () => {
      const markNotificationsReadMock = jest.fn();

      const props = {
        notification: mockSingleAtlassifyNotification,
        account: mockAtlassianCloudAccount,
      };

      render(
        <AppContext.Provider
          value={{
            settings: {
              ...mockSettings,
              delayNotificationState: true,
            },
            markNotificationsRead: markNotificationsReadMock,
            auth: mockAuth,
          }}
        >
          <NotificationRow {...props} />
        </AppContext.Provider>,
      );

      fireEvent.click(screen.getByTestId('notification-row'));

      expect(links.openNotification).toHaveBeenCalledTimes(1);
      expect(markNotificationsReadMock).toHaveBeenCalledTimes(1);
    });

    it('should mark a notification as read', () => {
      const markNotificationsReadMock = jest.fn();

      const props = {
        notification: mockSingleAtlassifyNotification,
        account: mockAtlassianCloudAccount,
      };

      render(
        <AppContext.Provider
          value={{
            settings: mockSettings,
            markNotificationsRead: markNotificationsReadMock,
          }}
        >
          <NotificationRow {...props} />
        </AppContext.Provider>,
      );

      fireEvent.click(screen.getByTestId('notification-mark-as-read'));

      expect(markNotificationsReadMock).toHaveBeenCalledTimes(1);
    });

    it('should mark a notification as unread', () => {
      const markNotificationsUnreadMock = jest.fn();

      const props = {
        notification: {
          ...mockSingleAtlassifyNotification,
          readState: 'read' as ReadState,
        },
        account: mockAtlassianCloudAccount,
      };

      render(
        <AppContext.Provider
          value={{
            settings: mockSettings,
            markNotificationsUnread: markNotificationsUnreadMock,
          }}
        >
          <NotificationRow {...props} />
        </AppContext.Provider>,
      );

      fireEvent.click(screen.getByTestId('notification-mark-as-unread'));

      expect(markNotificationsUnreadMock).toHaveBeenCalledTimes(1);
    });
  });
});
