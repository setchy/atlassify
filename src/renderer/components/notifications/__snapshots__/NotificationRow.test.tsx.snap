// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/components/notifications/NotificationRow.tsx should render itself & its children - group by date 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="group flex px-2 py-2 border-b border-notifications-light dark:border-notifications-dark hover:bg-notifications-light dark:hover:bg-notifications-dark"
        id="138661096"
      >
        <div
          class="mr-3 flex items-center justify-center"
        >
          <div
            class="css-2k1qcz"
          >
            <div
              role="presentation"
            >
              <div
                aria-labelledby="uid2"
                role="img"
                style="display: inline-block; position: relative; outline: 0;"
              >
                <span
                  class="css-1hqd2wf"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    class="css-13ep12v"
                    data-vc="avatar-image"
                    src="https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/123"
                    style="border-radius: 50%;"
                  />
                </span>
                <span
                  hidden=""
                  id="uid2"
                >
                  atlassify-app
                </span>
              </div>
            </div>
            <div
              role="presentation"
            >
              <span
                aria-hidden="true"
                class="css-1wits42"
                data-vc="icon-undefined"
                style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
              >
                <svg
                  height="24"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M6 12.235v6.446c0 .537.45.977 1 .977s1-.44 1-.977v-5.778c1.17-.341 2.389-.045 3.768.29.982.238 2.036.495 3.13.495a5.7 5.7 0 0 0 2.547-.567 1 1 0 0 0 .563-.9V5.466a.998.998 0 0 0-1.437-.9c-1.345.654-2.731.317-4.331-.071-1.729-.42-3.687-.895-5.678.072A1 1 0 0 0 6 5.466zm3.111-1.48c1.094 0 2.148.256 3.129.495 1.381.335 2.6.63 3.768.289V6.836c-1.488.27-2.93-.08-4.24-.398-1.379-.335-2.598-.632-3.768-.29v4.704a6 6 0 0 1 1.111-.097"
                    fill="currentcolor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex flex-1 cursor-pointer text-wrap"
          data-testid="notification-row"
          role="button"
          tabindex="0"
        >
          <div
            class="css-2mvdbp"
          >
            <div
              class="css-k5cxwk"
            >
              <span
                class="css-b0hel9"
              >
                #103: chore(deps): update dependency eslint
              </span>
                
              <em
                class="css-1ot9q9i"
              >
                3 years ago
              </em>
            </div>
            <div
              class="css-k5cxwk"
            >
              <div
                class="css-1w1rt7h"
              >
                <div
                  aria-labelledby="uid4"
                  role="img"
                  style="display: inline-block; position: relative; outline: 0;"
                >
                  <span
                    class="css-122fgfa"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      class="css-13ep12v"
                      data-vc="avatar-image"
                      src="https://bbc-object-storage--frontbucket.us-east-1.prod.public.atl-paas.net/80907f89f58b/img/favicon.ico"
                      style="border-radius: 2px;"
                    />
                  </span>
                  <span
                    hidden=""
                    id="uid4"
                  >
                    #103: chore(deps): update dependency eslint
                  </span>
                </div>
                <span
                  class="css-lym9kh"
                >
                  #103: chore(deps): update dependency eslint
                </span>
              </div>
              <div
                class="css-1yqbixt"
              >
                <div
                  class="css-f86pb"
                >
                  <span
                    aria-label="Bitbucket"
                    class="css-5vjqu0"
                    role="img"
                    style="--logo-color: inherit; --logo-fill: currentColor;"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="32"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      
      
                      <path
                        d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                        fill="#1868DB"
                      />
                      
    
                    </svg>
                  </span>
                  <span
                    class="css-lym9kh"
                  >
                    myorg/notifications-test
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1barwao"
        >
          <div
            role="presentation"
          >
            <button
              class="css-a8hcye"
              data-testid="notification-mark-as-read"
              type="button"
            >
              <span
                class="css-1spmf3f"
              >
                <span
                  aria-hidden="true"
                  class="css-1wits42"
                  data-vc="icon-undefined"
                  style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                >
                  <svg
                    height="24"
                    role="presentation"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      fill="currentcolor"
                      fill-rule="evenodd"
                      r="5"
                    />
                  </svg>
                </span>
                <span
                  class="css-b5o75w"
                >
                  Mark as read
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="group flex px-2 py-2 border-b border-notifications-light dark:border-notifications-dark hover:bg-notifications-light dark:hover:bg-notifications-dark"
      id="138661096"
    >
      <div
        class="mr-3 flex items-center justify-center"
      >
        <div
          class="css-2k1qcz"
        >
          <div
            role="presentation"
          >
            <div
              aria-labelledby="uid2"
              role="img"
              style="display: inline-block; position: relative; outline: 0;"
            >
              <span
                class="css-1hqd2wf"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="css-13ep12v"
                  data-vc="avatar-image"
                  src="https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/123"
                  style="border-radius: 50%;"
                />
              </span>
              <span
                hidden=""
                id="uid2"
              >
                atlassify-app
              </span>
            </div>
          </div>
          <div
            role="presentation"
          >
            <span
              aria-hidden="true"
              class="css-1wits42"
              data-vc="icon-undefined"
              style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
            >
              <svg
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M6 12.235v6.446c0 .537.45.977 1 .977s1-.44 1-.977v-5.778c1.17-.341 2.389-.045 3.768.29.982.238 2.036.495 3.13.495a5.7 5.7 0 0 0 2.547-.567 1 1 0 0 0 .563-.9V5.466a.998.998 0 0 0-1.437-.9c-1.345.654-2.731.317-4.331-.071-1.729-.42-3.687-.895-5.678.072A1 1 0 0 0 6 5.466zm3.111-1.48c1.094 0 2.148.256 3.129.495 1.381.335 2.6.63 3.768.289V6.836c-1.488.27-2.93-.08-4.24-.398-1.379-.335-2.598-.632-3.768-.29v4.704a6 6 0 0 1 1.111-.097"
                  fill="currentcolor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex flex-1 cursor-pointer text-wrap"
        data-testid="notification-row"
        role="button"
        tabindex="0"
      >
        <div
          class="css-2mvdbp"
        >
          <div
            class="css-k5cxwk"
          >
            <span
              class="css-b0hel9"
            >
              #103: chore(deps): update dependency eslint
            </span>
              
            <em
              class="css-1ot9q9i"
            >
              3 years ago
            </em>
          </div>
          <div
            class="css-k5cxwk"
          >
            <div
              class="css-1w1rt7h"
            >
              <div
                aria-labelledby="uid4"
                role="img"
                style="display: inline-block; position: relative; outline: 0;"
              >
                <span
                  class="css-122fgfa"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    class="css-13ep12v"
                    data-vc="avatar-image"
                    src="https://bbc-object-storage--frontbucket.us-east-1.prod.public.atl-paas.net/80907f89f58b/img/favicon.ico"
                    style="border-radius: 2px;"
                  />
                </span>
                <span
                  hidden=""
                  id="uid4"
                >
                  #103: chore(deps): update dependency eslint
                </span>
              </div>
              <span
                class="css-lym9kh"
              >
                #103: chore(deps): update dependency eslint
              </span>
            </div>
            <div
              class="css-1yqbixt"
            >
              <div
                class="css-f86pb"
              >
                <span
                  aria-label="Bitbucket"
                  class="css-5vjqu0"
                  role="img"
                  style="--logo-color: inherit; --logo-fill: currentColor;"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="32"
                    viewBox="0 0 32 32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    
      
                    <path
                      d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                      fill="#1868DB"
                    />
                    
    
                  </svg>
                </span>
                <span
                  class="css-lym9kh"
                >
                  myorg/notifications-test
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1barwao"
      >
        <div
          role="presentation"
        >
          <button
            class="css-a8hcye"
            data-testid="notification-mark-as-read"
            type="button"
          >
            <span
              class="css-1spmf3f"
            >
              <span
                aria-hidden="true"
                class="css-1wits42"
                data-vc="icon-undefined"
                style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
              >
                <svg
                  height="24"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    fill="currentcolor"
                    fill-rule="evenodd"
                    r="5"
                  />
                </svg>
              </span>
              <span
                class="css-b5o75w"
              >
                Mark as read
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/notifications/NotificationRow.tsx should render itself & its children - group by products 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="group flex px-2 py-2 border-b border-notifications-light dark:border-notifications-dark hover:bg-notifications-light dark:hover:bg-notifications-dark"
        id="138661096"
      >
        <div
          class="mr-3 flex items-center justify-center"
        >
          <div
            class="css-2k1qcz"
          >
            <div
              role="presentation"
            >
              <div
                aria-labelledby="uid8"
                role="img"
                style="display: inline-block; position: relative; outline: 0;"
              >
                <span
                  class="css-1hqd2wf"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    class="css-13ep12v"
                    data-vc="avatar-image"
                    src="https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/123"
                    style="border-radius: 50%;"
                  />
                </span>
                <span
                  hidden=""
                  id="uid8"
                >
                  atlassify-app
                </span>
              </div>
            </div>
            <div
              role="presentation"
            >
              <span
                aria-hidden="true"
                class="css-1wits42"
                data-vc="icon-undefined"
                style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
              >
                <svg
                  height="24"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M6 12.235v6.446c0 .537.45.977 1 .977s1-.44 1-.977v-5.778c1.17-.341 2.389-.045 3.768.29.982.238 2.036.495 3.13.495a5.7 5.7 0 0 0 2.547-.567 1 1 0 0 0 .563-.9V5.466a.998.998 0 0 0-1.437-.9c-1.345.654-2.731.317-4.331-.071-1.729-.42-3.687-.895-5.678.072A1 1 0 0 0 6 5.466zm3.111-1.48c1.094 0 2.148.256 3.129.495 1.381.335 2.6.63 3.768.289V6.836c-1.488.27-2.93-.08-4.24-.398-1.379-.335-2.598-.632-3.768-.29v4.704a6 6 0 0 1 1.111-.097"
                    fill="currentcolor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex flex-1 cursor-pointer text-wrap"
          data-testid="notification-row"
          role="button"
          tabindex="0"
        >
          <div
            class="css-2mvdbp"
          >
            <div
              class="css-k5cxwk"
            >
              <span
                class="css-b0hel9"
              >
                #103: chore(deps): update dependency eslint
              </span>
                
              <em
                class="css-1ot9q9i"
              >
                3 years ago
              </em>
            </div>
            <div
              class="css-k5cxwk"
            >
              <div
                class="css-1w1rt7h"
              >
                <div
                  aria-labelledby="uid10"
                  role="img"
                  style="display: inline-block; position: relative; outline: 0;"
                >
                  <span
                    class="css-122fgfa"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      class="css-13ep12v"
                      data-vc="avatar-image"
                      src="https://bbc-object-storage--frontbucket.us-east-1.prod.public.atl-paas.net/80907f89f58b/img/favicon.ico"
                      style="border-radius: 2px;"
                    />
                  </span>
                  <span
                    hidden=""
                    id="uid10"
                  >
                    #103: chore(deps): update dependency eslint
                  </span>
                </div>
                <span
                  class="css-lym9kh"
                >
                  #103: chore(deps): update dependency eslint
                </span>
              </div>
              <div
                class="css-1yqbixt"
              >
                <div
                  class="css-f86pb"
                >
                  <span
                    aria-label="Bitbucket"
                    class="css-5vjqu0"
                    role="img"
                    style="--logo-color: inherit; --logo-fill: currentColor;"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="32"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      
      
                      <path
                        d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                        fill="#1868DB"
                      />
                      
    
                    </svg>
                  </span>
                  <span
                    class="css-lym9kh"
                  >
                    myorg/notifications-test
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1barwao"
        >
          <div
            role="presentation"
          >
            <button
              class="css-a8hcye"
              data-testid="notification-mark-as-read"
              type="button"
            >
              <span
                class="css-1spmf3f"
              >
                <span
                  aria-hidden="true"
                  class="css-1wits42"
                  data-vc="icon-undefined"
                  style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                >
                  <svg
                    height="24"
                    role="presentation"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      fill="currentcolor"
                      fill-rule="evenodd"
                      r="5"
                    />
                  </svg>
                </span>
                <span
                  class="css-b5o75w"
                >
                  Mark as read
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="group flex px-2 py-2 border-b border-notifications-light dark:border-notifications-dark hover:bg-notifications-light dark:hover:bg-notifications-dark"
      id="138661096"
    >
      <div
        class="mr-3 flex items-center justify-center"
      >
        <div
          class="css-2k1qcz"
        >
          <div
            role="presentation"
          >
            <div
              aria-labelledby="uid8"
              role="img"
              style="display: inline-block; position: relative; outline: 0;"
            >
              <span
                class="css-1hqd2wf"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="css-13ep12v"
                  data-vc="avatar-image"
                  src="https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/123"
                  style="border-radius: 50%;"
                />
              </span>
              <span
                hidden=""
                id="uid8"
              >
                atlassify-app
              </span>
            </div>
          </div>
          <div
            role="presentation"
          >
            <span
              aria-hidden="true"
              class="css-1wits42"
              data-vc="icon-undefined"
              style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
            >
              <svg
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M6 12.235v6.446c0 .537.45.977 1 .977s1-.44 1-.977v-5.778c1.17-.341 2.389-.045 3.768.29.982.238 2.036.495 3.13.495a5.7 5.7 0 0 0 2.547-.567 1 1 0 0 0 .563-.9V5.466a.998.998 0 0 0-1.437-.9c-1.345.654-2.731.317-4.331-.071-1.729-.42-3.687-.895-5.678.072A1 1 0 0 0 6 5.466zm3.111-1.48c1.094 0 2.148.256 3.129.495 1.381.335 2.6.63 3.768.289V6.836c-1.488.27-2.93-.08-4.24-.398-1.379-.335-2.598-.632-3.768-.29v4.704a6 6 0 0 1 1.111-.097"
                  fill="currentcolor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex flex-1 cursor-pointer text-wrap"
        data-testid="notification-row"
        role="button"
        tabindex="0"
      >
        <div
          class="css-2mvdbp"
        >
          <div
            class="css-k5cxwk"
          >
            <span
              class="css-b0hel9"
            >
              #103: chore(deps): update dependency eslint
            </span>
              
            <em
              class="css-1ot9q9i"
            >
              3 years ago
            </em>
          </div>
          <div
            class="css-k5cxwk"
          >
            <div
              class="css-1w1rt7h"
            >
              <div
                aria-labelledby="uid10"
                role="img"
                style="display: inline-block; position: relative; outline: 0;"
              >
                <span
                  class="css-122fgfa"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    class="css-13ep12v"
                    data-vc="avatar-image"
                    src="https://bbc-object-storage--frontbucket.us-east-1.prod.public.atl-paas.net/80907f89f58b/img/favicon.ico"
                    style="border-radius: 2px;"
                  />
                </span>
                <span
                  hidden=""
                  id="uid10"
                >
                  #103: chore(deps): update dependency eslint
                </span>
              </div>
              <span
                class="css-lym9kh"
              >
                #103: chore(deps): update dependency eslint
              </span>
            </div>
            <div
              class="css-1yqbixt"
            >
              <div
                class="css-f86pb"
              >
                <span
                  aria-label="Bitbucket"
                  class="css-5vjqu0"
                  role="img"
                  style="--logo-color: inherit; --logo-fill: currentColor;"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="32"
                    viewBox="0 0 32 32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    
      
                    <path
                      d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                      fill="#1868DB"
                    />
                    
    
                  </svg>
                </span>
                <span
                  class="css-lym9kh"
                >
                  myorg/notifications-test
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1barwao"
      >
        <div
          role="presentation"
        >
          <button
            class="css-a8hcye"
            data-testid="notification-mark-as-read"
            type="button"
          >
            <span
              class="css-1spmf3f"
            >
              <span
                aria-hidden="true"
                class="css-1wits42"
                data-vc="icon-undefined"
                style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
              >
                <svg
                  height="24"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    fill="currentcolor"
                    fill-rule="evenodd"
                    r="5"
                  />
                </svg>
              </span>
              <span
                class="css-b5o75w"
              >
                Mark as read
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
